### PR TITLE
SD-122 - update drone.yml to version 1

### PR DESCRIPTION
**Pull request by SarahJaneLuff**
**Jira Ticket:** https://collaboration.homeoffice.gov.uk/jira/browse/SD-122

**Changes:** Updated drone.yaml file to version 1

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -1,116 +1,182 @@
...
-    image: docker:18.03.1-ce
-    environment:
-      - DOCKER_HOST=tcp://172.17.0.1:2375
-    commands:
-      - docker build -t app-$${DRONE_COMMIT_SHA} .
-    when:
-      branch: master
-      event: push
+steps:
+- name: build_are
```
Comment (markwimpory) [inline .drone.yml:7]: I dont think we want this step

File: .drone.yml [hunk 2]
```diff
@@ -1,116 +1,182 @@
...
-  build_acceptance:
-    image: docker:18.03.1-ce
-    environment:
-      - DOCKER_HOST=tcp://172.17.0.1:2375
-    commands:
-      - docker build -f Dockerfile-acceptance -t acceptance-$${DRONE_COMMIT_SHA} .
-    when:
-      branch: master
-      event: push
+- name: build_are_1
```
Comment (markwimpory) [inline .drone.yml:21]: rename this step

File: .drone.yml [hunk 3]
```diff
@@ -1,116 +1,182 @@
...
-    environment:
-      - DOCKER_HOST=tcp://172.17.0.1:2375
-    commands:
-      - docker run -d --name=redis-$${DRONE_COMMIT_SHA} redis
-      - docker run -d -e NODE_ENV=ci -e DISABLE_CSP=true --name=app-$${DRONE_COMMIT_SHA} --net=container:redis-$${DRONE_COMMIT_SHA} app-$${DRONE_COMMIT_SHA}
-      - docker run --rm --net=container:app-$${DRONE_COMMIT_SHA} acceptance-$${DRONE_COMMIT_SHA}
-      - docker rm -vf "app-$${DRONE_COMMIT_SHA}" "redis-$${DRONE_COMMIT_SHA}"
-    when:
-      event: pull_request
+- name: build_acceptance
```
Comment (markwimpory) [inline .drone.yml:36]: Again I dont think this is needed...its wasteful.  lets just build and test of master and feature branches

File: .drone.yml (hunk 1 above)
Comment (SarahJaneLuff) [reply .drone.yml:7]: removed this step

File: .drone.yml (hunk 2 above)
Comment (SarahJaneLuff) [reply .drone.yml:21]: renamed this to build_are

File: .drone.yml (hunk 3 above)
Comment (SarahJaneLuff) [reply .drone.yml:36]: removed this step, and done build and test of master and feature branches

File: .drone.yml [hunk 4]
```diff
@@ -1,116 +1,122 @@
...
-      event: push
+steps:
+- name: build_are
+  pull: if-not-exists
+  image: docker:dind
+  commands:
+  - n=0; while [ "$n" -lt 60 ] && [ ! -e /var/run/docker.sock ]; do n=$(( n + 1 )); sleep 1; done
+  - docker build -t app-$${DRONE_COMMIT_SHA} .
+  environment:
+    IMAGE_NAME: app-$${DRONE_COMMIT_SHA}
```
Comment (markwimpory) [inline .drone.yml:14]: This can be removed as its not used as far as I can see

Comment (SarahJaneLuff) [reply .drone.yml]: removed environment, and variable

**Pull request merged**